src/UserPage/index.tsx

```diff
@@ -1,39 +1,237 @@
+    type: 'twitter' | 'website' | 'instagram';
+    getUrl: (handle?: string) => string;
+    getLinkLabel: (handle?: string) => string;
+    labelKey: TFuncKey<'translation'>;
+    placeholerKey: TFuncKey<'translation'>;
+    labelHintKey: TFuncKey<'translation'>;
+}[] = [
+    {
+        type: 'instagram',
+        getUrl: (handle?: string) => `https://www.instagram.com/${handle}`,
+        getLinkLabel: (handle?: string) => `@${handle}`,
+        labelKey: 'userPage.socialLabels.instagram.label',
+        placeholerKey: 'userPage.socialLabels.instagram.placeholder',
+        labelHintKey: 'userPage.socialLabels.instagram.inputHint',
+    },
+    {
+        type: 'twitter',
+        getUrl: (handle?: string) => `https://twitter.com/${handle}`,
+        getLinkLabel: (handle?: string) => `@${handle}`,
+        labelKey: 'userPage.socialLabels.twitter.label',
+        placeholerKey: 'userPage.socialLabels.twitter.placeholder',
+        labelHintKey: 'userPage.socialLabels.twitter.inputHint',
+    },
+    {
+        type: 'website',
+        getUrl: (handle?: string) => `${handle}`,
+        getLinkLabel: (handle?: string) => `${removeHttpsWwwPageParams(handle)}`,
+        labelKey: 'userPage.socialLabels.website.label',
+        placeholerKey: 'userPage.socialLabels.website.placeholder',
+        labelHintKey: 'userPage.socialLabels.website.inputHint',
+    },
+];
 
 export default function UserPage() {
-    const user = useUser()!;
-    const userSettings = useUserSettings()!;
+    const { userId } = useParams<{ userId: string }>();
+    const loggedInUser = useUser()!;
     const { t } = useTranslation();
+    const [isEditing, setIsEditing] = useState(false);
+    const user = useDocument(collections.users.doc(userId));
+    const loggedInUserIsCurrentUser = loggedInUser.id === userId;
 
-    const onChange = useCallback(
-        (update: Partial<UserSettings>) => collections.userSettings.doc(user.id).set({ ...userSettings, ...update }),
-        [user.id, userSettings]
+    const edit = loggedInUserIsCurrentUser
+        ? () => {
+              setIsEditing(true);
+          }
+        : undefined;
+
+    return (
+        <>
+            <Header topHeading={[{ inner: t('userPage.title') }]} subLine={<UserBio user={user} edit={edit} />}>
+                {user.displayName}
+            </Header>
+            <Columns>
+                <UserInfo user={user} edit={edit} />
+                {loggedInUserIsCurrentUser && <EditUserSettings user={loggedInUser} />}
+            </Columns>
+            {isEditing && <EditUserInfo user={user} onClose={() => setIsEditing(false)} />}
+        </>
+    );
+}
+
+function UserInfo({ user, edit }: { user: User; edit?: () => void }) {
+    const { t } = useTranslation();
+
+    const socialMediaProfiles = USER_LINKS.filter(({ type }) => user.socialMediaProfiles?.[type]).map(
+        ({ type, getUrl, getLinkLabel }) => {
+            return { type, getUrl, getLinkLabel };
+        }
+    );
```

Comment:
```suggestion
    const socialMediaProfiles = USER_LINKS.filter(({ type }) => !!user.socialMediaProfiles?.[type]);
```
I think the extra map can go